samples/unit-tests/maps/series-map/demo.js

```diff
@@ -61,92 +61,99 @@ QUnit.test('Mapping of joinBy with data', assert => {
+                        },
+                        {
+                            name: 'New South Wales',
+                            value: 9,
+                            path:
+                                'M686,505,760,511,815,516,876,523,884,515,890,514,895,515,906,515,910,517,913,518,918,519,922,522,924,524,928,528,928,533,932,534,936,532,939,527,943,530,949,530,951,525,950,520,963,515,968,516,973,519,986,516,985,529,979,542,973,557,966,572,966,582,962,592,956,605,951,611,947,617,944,624,926,636,912,653,907,664,901,674,897,689,890,696,881,706,876,718,872,731,869,749,866,752,832,730,833,723,831,717,831,709,822,705,811,706,801,704,789,704,776,699,767,695,759,697,759,700,759,703,758,704,754,702,749,692,742,686,734,681,732,677,728,672,730,664,725,661,716,658,711,662,708,654,707,649,698,642,693,641,688,643,678,639z',
+                            middleY: 0.4
+                        },
+                        {
+                            name: 'A.C.T.',
+                            value: 2,
+                            path:
+                                'M848,696,853,689,859,689,861,695,856,701,854,711,849,706,848,701z'
+                        },
+                        {
+                            selected: true,
+                            name: 'Victoria',
+                            value: 3,
+                            path:
+                                'M677,643,670,753,683,762,689,758,698,761,711,769,722,777,744,766,749,762,755,753,758,758,755,765,754,769,766,776,774,780,776,784,795,780,811,765,825,761,842,760,858,761,864,755,829,732,828,728,829,722,827,720,826,712,822,710,816,709,811,711,804,709,800,707,794,710,789,708,774,703,769,701,763,701,762,706,759,708,754,707,749,700,745,692,737,686,730,684,726,677,726,669,724,664,719,663,715,662,711,665,707,659,704,656,704,651,695,646,691,647,685,646z',
+                            middleY: 0.6
+                        }
+                    ],
+                    name: 'Random data'
+                }
+            ]
+        }),
+        series = chart.series[0];
 
-    chart = $('#container').highcharts();
-    chart.series[0].data[4].select();
-    chart.series[0].data[0].update(100);
+    series.points[4].select();
+    series.points[0].update(100);
 
     assert.strictEqual(
         chart.series[0].data[4].graphic.attr('fill'),
         '#ff00ff',
         'Point use selected color.'
     );
+
+    series.points[0].setState('hover', false);
+    series.points[0].setState('normal', false);
+
+    assert.strictEqual(
+        series.points[0].graphic['stroke-width'],
+        undefined,
+        'Map border should be set to initial after changing state (#18137).'
+    );
```

Comment:
It would be good to check if the hover worked because if it fails the test will pass regardless of whether the actually tested functionality is working or not.

Reply:
Added test one more assert.